src/lib.rs

```diff
@@ -363,12 +363,24 @@ impl Check {
             RustdocSource::Root(project_root) => {
                 let metadata = manifest_metadata(project_root)?;
                 let selected = self.scope.selected_packages(&metadata);
+
+                let rustc_version_1_65 = rustc_version::Version::new(1, 65, 0);
+
                 selected
                     .iter()
                     .map(|selected| {
                         let crate_name = &selected.name;
                         let version = &selected.version;
 
+                        let rust_version = rustc_version::version()
+                            .expect("the rust version could not be determined !");
+                        if rust_version < rustc_version_1_65 {
+                            anyhow::bail!(
+                                "rustc version is not high enough: >=1.65.0 needed, got {}\n\nHELP: to run with the latest rustc version, use `rustup update stable && cargo +stable semver-checks ...`",
+                                rust_version
+                            );
+                        }
```

Comment:
I think this logic should be outside of this `match` so that it applies to all cases, not just this most-common one.

And then we'll probably want to explicitly skip the check if both current and baseline rustdoc files are given explicitly instead of being generated through the tool, since then the installed Rust version doesn't matter.

Reply:
Done!